Odata-docs/webapi-8/fundamentals/property-routing.md

```diff
@@ -0,0 +1,1005 @@
+```csharp
+public ActionResult GetBillingAddress([FromRoute] int key)
+{
+    var customer = customers.SingleOrDefault(d => d.Id.Equals(key));
+
+    if (customer == null)
+    {
+        return NotFound();
+    }
+
+    return Ok(customer.BillingAddress);
+}
+```
+
+The following JSON payload shows the expected response:
+```json
+{
+    "@odata.context": "http://localhost:5000/odata/$metadata#Customers(1)/BillingAddress",
+    "Street": "Street 1A"
+}
+```
+
+### Retrieving a property initialized to an instance of a derived type
+The route templates for this request are:
+- `GET ~/{entityset}({key})/{property}/{cast}`
+- `GET ~/{entityset}/{key}/{property}/{cast}`
+- `GET ~/{singleton}/{property}/{cast}`
+
+The following request returns `BillingAddress` property (_on customer 2_) initialized to an instance of `PostalAddress`:
+```http
+GET http://localhost:5000/odata/Customers(2)/BillingAddress/PropertyRouting.Models.PostalAddress
+```
+
+For the above request to be conventionally-routed, a controller action named `GetBillingAddressOfPostalAddress` that accepts the key parameter is expected:
+```csharp
+public ActionResult GetBillingAddressOfPostalAddress([FromRoute] int key)
+{
+    var customer = customers.SingleOrDefault(d => d.Id.Equals(key));
+
+    if (!(customer?.BillingAddress is PostalAddress billingAddress))
+    {
+        return NotFound();
+    }
+
+    return Ok(billingAddress);
+}
+```
+
+The following JSON payload shows the expected response:
+```json
+{
+    "@odata.context": "http://localhost:5000/odata/$metadata#Customers(2)/BillingAddress/PropertyRouting.Models.PostalAddress",
+    "Street": "2A",
+    "PostalCode": "14030"
+}
+```
+
+It is important to note that the controller action from [retrieving a property](#retrieving-a-property) section can adequately handle request for `BillingAddress` property that is initialized to an instance of `PostalAddress`. The request would look at follows:
+```http
+GET http://localhost:5000/odata/Customers(2)/BillingAddress
+```
+
+The following JSON payload shows the expected response:
+```json
+{
+    "@odata.context": "http://localhost:5000/odata/$metadata#Customers(2)/BillingAddress",
+    "Street": "2A",
+    "PostalCode": "14030"
+}
+```
+
+The cast segment is useful in explicitly expressing the specific type of instance expected on the property. If the instance is not of the expected type [a `404 Not Found` response should be returned](http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part2-url-conventions.html#sec_AddressingDerivedTypes).
+
+### Retrieving a property declared on a derived type
+The route templates for this request are:
+- `GET ~/{entityset}({key})/{cast}/{property}`
+- `GET ~/{entityset}/{key}/{cast}/{property}`
+- `GET ~/{singleton}/{cast}/{property}`
+
+The following request returns `RegisteredAddress` property on enterprise customer 4:
```

Comment:
I think it may make the example clearer if you point out here that `RegisteredAddress` is declared in the `EnterpriseCustomer` class but not in the parent `Customer` class. I know that can be inferred from the model classes, but it's possible the reader didn't notice it or doesn't have that context by the time they get to this section.